### ISSUE TITLE
500 error on v3.6.4 when using imgix

### ISSUE THREAD

**Issue by dave-stillandnimble** [spacecatninja/craft-imager-x#175]
### I'm submitting a... 

- [X ] bug report
- [ ] feature request
- [ ] general question

### Steps to reproduce

1. Set the image transformer to imgix
2. Observer error when trying to do image transform in twig
3. ...

### Description
 
When upgrading from 3.6.2 to 3.6.4 I now get 500 errors for image transforms when the image transformer is set to imgix. When set to craft everything works as expected. 

### Displayed errors, stack trace, relevant logs  

```
Undefined index: h   in /Users/../vendor/spacecatninja/imager-x/src/transformers/ImgixTransformer.php
// Assume that the reset of the values left in the transform object is Imgix specific 
        foreach ($transform as $key => $val) {
            $r[$key] = $val;
        }
 
        // If allowUpscale is disabled, use max-w/-h instead of w/h
        if (isset($r['fit']) && !$config->getSetting('allowUpscale', $transform)) {
            if ($r['fit'] === 'crop' && $image instanceof Asset && ($r['w'] > $image->getWidth() || $r['h'] > $image->getHeight())) {
                $sourceRatio = $image->getHeight() / $image->getWidth();
                $transformRatio = (int)$r['h'] / (int)$r['w'];
 
                if ($sourceRatio > $transformRatio) {
                    $r['w'] = $image->getWidth();
                    $r['h'] = $image->getWidth() * $transformRatio;
                } else {
                    $r['h'] = $image->getHeight();
                    $r['w'] = $image->getHeight() / $transformRatio;
                }
            }

```

### Additional info

- Imager version & edition:3.6.4
- Imager transformer:imgix
- Craft version: 3.7.55.2
- PHP version:7.4.12
- Image driver & version:

**Comment by aelvan** [comment]
Probably same as https://github.com/spacecatninja/craft-imager-x/issues/173, only for Craft 3. 🤦‍♂️ Can you share your template code, just to make sure it's the same. (Ie, you're using `mode` set to `crop`, and only one dimension?)

**Comment by dave-stillandnimble** [comment]
Yep that sounds about right, my image macro code is:

```
{% import '_partials/_macros' as m %}

{% set defaultOptions = {
    image: null,
    sizes: [  
        { width: 1920 },
        { width: 1366 },
        { width: 768 },
        { width: 360 },
    ],
    minWidths: [
        { minWidth: 1280, fraction: 1/3},
        { minWidth: 1024, fraction: 1/2},
        { minWidth: 768,  fraction: 1/2},
        { minWidth: 568,  fraction: 1/2},
        { minWidth: 0,    fraction: 1/2},
    ],
    maxWidth: 1280,
    allowUpscale: false,
    alt: false,
    format: 'jpg',
    interlace: true,
    lazy: false,
    mode: 'crop',
    position: false,
    objectPosition: false,
    quality: 80,
    ratio: false, 
    class: ''
} %}



{# Merge Attr with Defaults #}
{% set options = options ? defaultOptions|merge(options) : defaultOptions %}

{% if options.image %}   
    {# Set Ratio #}
    {% if options.ratio %}
      {% set ratio = options.ratio|split(':') %}
      {% set base64Ratio = options.ratio|split(':') %}
      {% set ratio = ratio|first/ratio|last %}
    {% else %}
      {% set imageSize = options.image.width ~ ':' ~ options.image.height %}
      {% set base64Ratio = imageSize|split(':') %}
      {% set ratio = '' %}
    {% endif %}

    {# Set Position #}
    {% if options.position %}
      {% set position = options.position %}
    {% elseif options.image.getFocalPoint() %}
      {% set position = options.image.getFocalPoint() %}
      {% set position = position.x * 100 ~ '% ' ~ position.y * 100 ~'%' %}
    {% else %}
      {% set position = '50% 50%' %}
    {% endif %}

    {# Define global variables #}
    {% set imageSettings = {
        allowUpscale: options.allowUpscale,
        format: options.format,
        interlace: options.interlace,
        jpegQuality: options.quality,
        mode: options.mode,
        position: position,
        ratio: ratio,
    } %}

    {# Image #}
    {% set images = craft.imager.transformImage(options.image, options.sizes, imageSettings) %}

    {% set minWidths = options.minWidths|multisort('minWidth', sortFlag=SORT_NUMERIC, direction=SORT_DESC) %}
    
    {% set sizes %}
        {% for item in minWidths %}
            {% if loop.first %}
                {% if options.maxWidth %}
                    (min-width:{{item.minWidth}}px) {{item.fraction * options.maxWidth}}px,
                {% else %}
                    (min-width:{{item.minWidth}}px) {{item.fraction * 100}}vw,
                {% endif %}
            {% elseif loop.last %}
                {% if item.minWidth == 0 %}
                    {{item.fraction * 100}}vw
                {% else %}
                    (min-width:{{item.minWidth}}px) {{item.fraction * 100}}vw,
                {% endif %}
            {% else %}
                (min-width:{{item.minWidth}}px) {{item.fraction * 100}}vw,
            {% endif %}
            
        {% endfor %}
 
    {% endset %}

    <img sizes="{{sizes}}"       
        src="{{ images|last.url }}"
        srcset="{{ craft.imager.srcset(images) }}"
        class="{{ options.class }}"  
        width="{{base64Ratio|first}}"
        height="{{base64Ratio|last}}"
        {% if options.lazy %}loading="lazy"{% endif %}
        {% if options.objectPosition %}style="object-position: {{position}}"{% endif %}
        alt="{{ options.alt | default(options.image.altText) }}">
    
{% endif %}

```

and an example of it in use:

```
{{ m.element('image', {
    image: options.image,
    sizes: [  
        { width: 1366 },
        { width: 768 },
        { width: 360 },
    ],
    minWidths: [
        { minWidth: 1280, fraction: 1/2},
        { minWidth: 1024, fraction: 1/2},
        { minWidth: 768,  fraction: 1/2},
        { minWidth: 568,  fraction: 1/1},
        { minWidth: 0,    fraction: 1/1},
    ],
    maxWidth: 1536,
    allowUpscale: false,
    alt: false,
    format: 'png',
    interlace: true,
    lazy: false,
    mode: 'crop',
    position: false,
    quality: 80,
    ratio: false, 
    class: 'relative md:absolute md:inset-0 md:w-full md:h-full md:object-cover md:object-left-top'
}) }}
```

**Comment by aelvan** [comment]
Should be fixed in 3.6.5
